pkg/subscriptionmanager/factory.go

```diff
@@ -43,3 +50,18 @@ func (f Factory) NewJetStreamManager(eventing v1alpha1.Eventing, natsConfig env.
 	return jetstream.NewSubscriptionManager(f.k8sRestCfg, natsConfig.ToECENVNATSConfig(eventing),
 		f.metricsAddress, f.metricsCollector, f.logger)
 }
+
+func (f Factory) NewEventMeshManager(eventing v1alpha1.Eventing) (ecsubscriptionmanager.Manager, error) {
+	if err := setUpEventMeshSubManagerEnvironment(eventing); err != nil {
+		return nil, err
+	}
+	return eventmesh.NewSubscriptionManager(f.k8sRestCfg, f.metricsAddress, f.resyncPeriod, f.logger), nil
+}
+
+func setUpEventMeshSubManagerEnvironment(eventing v1alpha1.Eventing) error {
+	// Set the EVENT_TYPE_PREFIX env variable as it is read in the NewSubscriptionManager() function.
+	if err := os.Setenv("EVENT_TYPE_PREFIX", eventing.Spec.Backend.Config.EventTypePrefix); err != nil {
```

Comment:
Can you open an issue to improve this later? Its does not seem to be a good idea to set Environment variables at different places in code, atleast there should be a single place where all the ENVs for being defined, otherwise it would be difficult later to debug from where different values are coming from.

Reply:
Ok, I'll create an issue for this.